core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -96,6 +96,11 @@
         Dynamic,
 
         /**
+         * mark this setting as final (the value cannot be updated)
```

Comment:
yeah just put this explain part in the comment. we can be explicit and say that it also prevents updates if the index is closed